public/app/features/alerting/unified/components/rules/RulesTable.tsx

```diff
@@ -63,9 +63,19 @@ export const RulesTable = ({
 
   const { pageItems, page, numberOfPages, onPageChange } = usePagination(rules, 1, DEFAULT_PER_PAGE_PAGINATION);
 
-  const { result: rulesWithRulerDefinitions, status: rulerRulesLoadingStatus } = useLazyLoadRulerRules(pageItems);
+  const [lazyLoadRules, { result: rulesWithRulerDefinitions, status: rulerRulesLoadingStatus }] =
+    useLazyLoadRulerRules(pageItems);
+  const isLoadingRulerGroup = useMemo(
+    () =>
+      !rulerRulesLoadingStatus || rulerRulesLoadingStatus === 'loading' || rulerRulesLoadingStatus === 'not-executed',
+    [rulerRulesLoadingStatus]
+  );
```

Comment:
Was attempting to make the loading work correctly but it flicks between the actions rendering partly, then more buttons, then as a skeleton, then finally back to rendering properly 🤷 Think something strange is going on (but this case only happens on first page load if you're going _straight_ to the state list view)